File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        boolean[] prime = new boolean[3001];
        
        prime[0] = true;
        prime[1] = true;
        
        for(int i=2; i<=60; i++) {
            if(prime[i]) continue;
            
            for(int j = i*i; j <= 3000; j += i) {
                prime[j] = true;
            }
        }
        
        int length = nums.length;
        int answer = 0;
        for(int i=0; i<length-2; i++) {
            for(int j=i+1; j<length-1; j++) {
                for(int k=j+1; k<length; k++) {
                    if(!prime[nums[i] + nums[j] + nums[k]])
                        answer++;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<int> nums) {
    int max = 3000;
    
    vector<bool> isPrime(max + 1, true);
    isPrime[0] = false;
    isPrime[1] = false;

    for (int i=2; i*i<=max; i++) {
        if (!isPrime[i]) continue;
        
        for (int j=i*i; j<=max; j+=i) {
            isPrime[j] = false;
        }
    }

    int answer = 0;
    int n = nums.size();
    for (int i=0; i<n; i++) {
        for (int j=i+1; j<n; j++) {
            for (int k=j+1; k<n; k++) {
                if (isPrime[nums[i] + nums[j] + nums[k]]) {
                    answer++;
                }
            }
        }
    }
    
    return answer;
}